x-pack/legacy/plugins/watcher/public/legacy.ts

```diff
@@ -0,0 +1,147 @@
+
+/* Legacy UI imports */
+import { npSetup, npStart } from 'ui/new_platform';
+import routes from 'ui/routes';
+import { management, MANAGEMENT_BREADCRUMB } from 'ui/management';
+// @ts-ignore
+import { TimeBuckets } from 'ui/time_buckets';
+// @ts-ignore
+import { xpackInfo } from 'plugins/xpack_main/services/xpack_info';
+/* Legacy UI imports */
+
+import { plugin } from './np_ready';
+import { PLUGIN } from '../common/constants';
+import { LICENSE_STATUS_INVALID, LICENSE_STATUS_UNAVAILABLE } from '../../../common/constants';
+import { manageAngularLifecycle } from './manage_angular_lifecycle';
+
+const template = `<kbn-management-app section="elasticsearch/watcher">
+  <div id="watchReactRoot"></div>
+</kbn-management-app>`;
+
+let elem: HTMLElement;
+let mountApp: () => AppUnmount | Promise<AppUnmount>;
+let unmountApp: AppUnmount | Promise<AppUnmount>;
+routes.when('/management/elasticsearch/watcher/:param1?/:param2?/:param3?/:param4?', {
+  template,
+  controller: class WatcherController {
+    constructor($injector: any, $scope: any) {
+      const $route = $injector.get('$route');
+      const licenseStatus = xpackInfo.get(`features.${PLUGIN.ID}`);
+      const shimCore: CoreSetup = {
+        ...npSetup.core,
+        application: {
+          ...npSetup.core.application,
+          register(app: App): void {
+            mountApp = () =>
+              app.mount(npStart as any, {
+                element: elem,
+                appBasePath: '/management/elasticsearch/watcher/',
+              });
+          },
+        },
+      };
+
+      // clean up previously rendered React app if one exists
+      // this happens because of React Router redirects
+      if (elem) {
+        ((unmountApp as unknown) as AppUnmount)();
+      }
+
+      $scope.$$postDigest(() => {
+        elem = document.getElementById('watchReactRoot')!;
+        const instance = plugin();
+        instance.setup(shimCore, {
+          ...(npSetup.plugins as typeof npSetup.plugins & { eui_utils: any }),
+          __LEGACY: {
+            MANAGEMENT_BREADCRUMB,
+            TimeBuckets,
+            licenseStatus,
+          },
+        });
+
+        instance.start(npStart.core, npStart.plugins);
+
+        (mountApp() as Promise<AppUnmount>).then(fn => (unmountApp = fn));
+
+        manageAngularLifecycle($scope, $route, elem);
+      });
+    }
+  } as any,
+  // @ts-ignore
+  controllerAs: 'watchRoute',
+});
+
+routes.defaults(/\/management/, {
+  resolve: {
+    watcherManagementSection: () => {
+      const watchesSection = management.getSection('elasticsearch/watcher');
+      const licenseStatus = xpackInfo.get(`features.${PLUGIN.ID}`);
+      const { status } = licenseStatus;
+
+      if (status === LICENSE_STATUS_INVALID || status === LICENSE_STATUS_UNAVAILABLE) {
+        return watchesSection.hide();
+      }
+
+      watchesSection.show();
+    },
+  },
+});
+
+management.getSection('elasticsearch').register('watcher', {
```

Comment:
@mattkime You asked me today about these sections. Could you collaborate with @jloleysens to avoid merge conflicts and duplication of work? Maybe you could suggest the changes you want to make be made in this PR instead?

Reply:
I'm fairly certain that these can be removed and here's proof - https://github.com/elastic/kibana/pull/52568

@jloleysens you can decide how you'd prefer to proceed with this, one PR or multiple.

Reply:
Hi @mattkime! I reckon if CI goes green on the latest commit I'll merge this branch, if there are issues resolving hmu!